src/build.rs

```diff
@@ -15,12 +23,40 @@ pub struct BuildDirectory {
     name: String,
 }
 
-impl BuildDirectory {
-    pub(crate) fn new(workspace: Workspace, name: &str) -> Self {
-        Self {
-            workspace,
-            name: name.into(),
-        }
+/// Builder for configuring builds in a [`BuildDirectory`](struct.BuildDirectory.html).
+pub struct Builder<'a> {
```

Comment:
Could we use a less generic name here? Maybe `BuildBuilder`?

Also, this type needs to be re-exported at the top of the crate, otherwise it won't show up in the docs:

![2019-09-16--09-07-20](https://user-images.githubusercontent.com/2299951/64939481-6639c480-d861-11e9-930e-5257ee391822.png)